### DISCUSSION TITLE
Changing the DisplayAlert button color

### DISCUSSION THREAD

**Discussion by Symbai** [dotnet/maui#6579] [Q&A]
I'm new to MAUI. I noticed the `Styles.xaml` file with color definitions in it. I changed all to my desires but unfortunately tons of controls are still using the default colors. For example the `DisplayAlert` button is still in purple. Same for the `Entry` underline and caret.

How can I change the colors for them?

**Comment by KarlSearl** [comment]
Not sure if you can do this with the display alert aspects.,  but there is as pop-up control in the Maui community toolkit.  I had a play with that, you have a lot of control over the style.

**Reply by Symbai** [reply]
It should be doable in MAUI without having to switch to a completely different third party popup. I only wanted to know if I have missed something or if it simply not implemented in MAUI yet

**Reply by 99067718** [reply]
Bruh that sucks.
Purple on top of gray is not readable at all...

**Reply by LaraSQP** [reply]
> Bruh that sucks. Purple on top of gray is not readable at all...

This is still a problem today.

Does anyone have a solution?

**Reply by Symbai** [reply]
@LaraSQP After some time I figured out how to change many of the "unchangeable" colors, such as Entry underline / caret etc. I don't know if this also changes the displayalert button color but you can give it a try: https://github.com/dotnet/maui/issues/9039

**Reply by LaraSQP** [reply]
Thank you very, very much for replying.

I will give it a try.

**Comment by mhrastegari** [comment]
If you only want to change the default purple color of all controls in android all you have to do is change the color variables in `YOURPROJECT\Platforms\Android\Resources\values\colors.xml` .

**Reply by LaraSQP** [reply]
Thank you very much. I was unaware of a `colors.xml` file specific to android.

Turns out that to change the purple `ok/cancel/etc` buttons in `DisplayAlert`, all that is needed is to modify `colorPrimary`.

You rock, man.

**Reply by mhrastegari** [reply]
Happy to help ❤️

**Reply by dnovaprove** [reply]
Thanks very much!

**Reply by blautben** [reply]
For anyone looking to change the line color on Entry/Picker, you're looking to change the `android:colorControlNormal` and `colorAccent` items in this file. `colorAccent` is the default value for `colorControlActivated`, so if you'd rather change it explicitly, that should work too.

**Comment by MauiUIui** [comment]
> For anyone looking to change the line color on Entry/Picker, you're looking to change the `android:colorControlNormal` and `colorAccent` items in this file. `colorAccent` is the default value for `colorControlActivated`, so if you'd rather change it explicitly, that should work too.

I know about `colorAccent` and I did change it to fit my needs. But where can I find `android:colorControlNormal` and `colorControlActivated`?

**Reply by mhrastegari** [reply]
you need to create a `styles.xml` file your self.

**Comment by hinnerkweiler** [comment]
Has anyone found a way to reflect screen modes for these controls? E.g. it is hard to find a color fpr the Alert Box "OK" that stands out equally on a black or white Background.

**Reply by appdevcanada** [reply]
That's why there are 2 colors.xml, one in values and the other in values-night. You just need to adjust the desired colours...

**Reply by hinnerkweiler** [reply]
oh... uhm... not in my MAUI?!? – I would have tried that. 
Just created a `values-night` folder and it magically comes into action on `dark-mode` indeed. Thank you!
From your wording it sounds as if that folder is normally generated automatically? If so (on Windows?) it may be a Maui-Bug on Mac that it does not?  

<img width="202" alt="Screenshot 2023-10-09 at 10 55 46" src="https://github.com/dotnet/maui/assets/18248862/bb911bd0-7dc2-4deb-bd29-eaad1164e75f">

**Reply by appdevcanada** [reply]
Yeah, sorry, but it was already built for my project... and using VS 2022 for Mac.

**Reply by hinnerkweiler** [reply]
ok, it than may have changed over time. The project was initially created with one of the latest RC's of 2022 Mac. However thanks again for the hint.

**Reply by Seenuvasanmohan** [reply]
how to apply for windows platform for applying button color for alert/actionsheet.

**Comment by CodingOctocat** [comment]
I noticed that the Alert button in the document is magenta, but mine is black.

https://learn.microsoft.com/en-us/dotnet/maui/user-interface/pop-ups?view=net-maui-8.0
![image](https://github.com/dotnet/maui/assets/7220248/59b38c6f-bb00-44cc-b9d4-54bf3c10f5ea)

**Reply by leonardbarbu** [reply]
That's because screenshots are from Xamarin when magenta was the default color on buttons. I think they didn't update the screenshots in the MAUI documentation.

Look there: https://learn.microsoft.com/en-us/previous-versions/xamarin/xamarin-forms/user-interface/pop-ups
